File: 현호/BOJ_5430.js
// 5430 : AC
// 문제 요약
// 커맨드에 따른 연산 결과를 반환하기
// R: 원소들을 뒤집는다, D: 가장 첫 번째 원소를 삭제한다

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];

const COMMAND = {
  reverse: "R",
  delete: "D",
};

rl.on("line", (line) => {
  input.push(line);
}).on("close", () => {
  const tryCount = Number(input.shift());
  solve(tryCount);
  process.exit();
});

const solve = (tryCount) => {
  for (let i = 0; i < tryCount; i += 1) {
    const commands = input.shift().split("");
    const arrLen = Number(input.shift());
    const shiftedArray = input.shift().slice(1, -1).split(",").map(Number);
    const arr = arrLen ? shiftedArray : [];
    let direction = 0;
    let flag = 1;

    for (let j = 0; j < commands.length; j += 1) {
      if (commands[j] === COMMAND.reverse) {
        direction = direction ? 0 : 1;
        continue;
      }
      if (commands[j] === COMMAND.delete && !arr.length) {
        flag = 0;
        console.log("error");
        break;
      }

      remove(arr, direction);
    }

    if (flag) {
      direction
        ? console.log(`[${arr.reverse().join(",")}]`)
        : console.log(`[${arr.join(",")}]`);
    }
  }
};

const remove = (arr, direction) => {
  direction ? arr.pop() : arr.shift();
};

File: 혜민/BOJ_5430.py
from collections import deque

def D_func(queue, R_count):
    if len(queue) == 0:
        return 0
    else:
        if R_count % 2 == 1:
            queue.pop()     #홀수면 오른쪽 pop
            return list(queue)
        else:
            queue.popleft()    #짝수면 왼쪽 pop
            return list(queue)

ans = []
T = int(input())
for _ in range(T):
    func = input()
    n = int(input())
    arr = list(input()[1:-1].split(',')) 
    R_count = 0

    queue = deque(arr)

    if n == 0:  #빈 배열이 들어왔을때 ['']이 상태로 인식이 되면, 길이가 0이 아닌 1로 인식되므로 빈 리스트로 초기화해줘야함
        queue = []

    for f in func:         
        if f == 'R':        #R인 경우엔 나중에 한번에 연산해야하기 떄문에 count만 하기 
            R_count += 1
        else:
            queue = D_func(queue, R_count)      #D인 경우에 D함수 호출
            if queue == 0:  
                break

    #함수 연산이 끝난 후 r_count 값에 따른 리버스 수행
    if R_count % 2 == 1:
        queue.reverse()

    #queue가 비어있다면 error 정답 리스트에 추가, 아니면 큐 추가 
    if queue == 0:
        ans.append('error')
    else:
        ans.append(queue)

for a in ans:
    if a == 'error':
        print(a)
    else:
        print("[" + ",".join(a) + "]")
